clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -232,6 +233,36 @@ public void close() {
         channelBuilder.close();
     }
 
+    @Override
+    public void closeGracefully() {
+        try {
+            long endTimeMs = -1;
```

Comment:
nit: Would it be clearer to initialize this to `time.milliseconds() + CLOSE_TIMEOUT_MS` and avoid the sentinel check? It doesn't seem too worthwhile to avoid the system call.